frontend/public/components/edit-yaml.jsx

```diff
@@ -35,8 +35,9 @@ const generateObjToLoad = (kind, templateName, namespace = 'default') => {
   return sampleObj;
 };
 
-const stateToProps = ({k8s}, {obj}) => ({
-  model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
```

Comment:
How does this fix the problem exactly? Didn't you just move this logic into a component method instead of `stateToProps`?

Reply:
The model is different depending on what is typed into the YAML, so we don't get the model until the user clicks save.